gunicorn/instrument/statsd.py

```diff
@@ -98,29 +100,22 @@ def access(self, resp, req, environ, request_time):
     # statsD methods
     # you can use those directly if you want
     def gauge(self, name, value):
-        try:
-            if self.sock:
-                self.sock.send("{0}{1}:{2}|g".format(self.prefix, name, value))
-        except Exception:
-            pass
+        self._sock_send("{0}{1}:{2}|g".format(self.prefix, name, value))
 
     def increment(self, name, value, sampling_rate=1.0):
-        try:
-            if self.sock:
-                self.sock.send("{0}{1}:{2}|c|@{3}".format(self.prefix, name, value, sampling_rate))
-        except Exception:
-            pass
+        self._sock_send("{0}{1}:{2}|c|@{3}".format(self.prefix, name, value, sampling_rate))
 
     def decrement(self, name, value, sampling_rate=1.0):
-        try:
-            if self.sock:
-                self.sock.send("{0){1}:-{2}|c|@{3}".format(self.prefix, name, value, sampling_rate))
-        except Exception:
-            pass
+        self._sock_send("{0){1}:-{2}|c|@{3}".format(self.prefix, name, value, sampling_rate))
 
     def histogram(self, name, value):
+        self._sock_send("{0}{1}:{2}|ms".format(self.prefix, name, value))
+
+    def _sock_send(self, msg):
         try:
+            if isinstance(msg, six.text_type):
+                msg = msg.encode("ascii")
```

Comment:
+1 let's merge now and handle the rest separately.
On Thu 30 Apr 2015 at 12:44 Berker Peksag <notifications@github.com> wrote:

> In gunicorn/instrument/statsd.py
> <https://github.com/benoitc/gunicorn/pull/1010#discussion_r29419336>:
>
> >          try:
> > +            if isinstance(msg, six.text_type):
> > +                msg = msg.encode("ascii")
>
> If we are going to add a function like this, we need to make it more
> general(e.g. you can pass different encodings and error handlers).
> util.to_bytestring is another alternative, but it uses the UTF-8 encoding
> by default. I'd say merge this now and create a new issue or PR to discuss
> details.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/benoitc/gunicorn/pull/1010/files#r29419336>.
>